packages/flutter/lib/src/rendering/object.dart

```diff
@@ -250,7 +250,7 @@ class PaintingContext extends ClipContext {
     // Create a layer for our child, and paint the child into it.
     if (child._needsPaint || !child._wasRepaintBoundary) {
       repaintCompositedChild(child, debugAlsoPaintedParent: true);
-    } else {
+    } else if (child._needsCompositedLayerUpdate) {
```

Comment:
The metrics recording stuff in the assert below needs to happen regardless of _needsCompositedLayerUpdate, though. It is just the updateLayerProperties that needs to be guarded, no?

Reply:
fixed